knowledge-base/grid-custom-cell-formatting-with-radspreadstreamprocessing.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Custom cell formatting of the exported file with RadSpreadStreamProcessing
+description: How to format the cells of the exported file with RadSpreadStreamProcessing?
+type: how-to
+page_title: Custom cell formatting of exported Excel files with RadSpreadStreamProcessing
+slug: grid-kb-custom-cell-formatting-with-radspreadstreamprocessing
+position: 
+tags: grid, custom, cell, format, radspreadstreamprocessing
+ticketid: 1598580
+res_type: kb
+---
+
+## Environment
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+I want to format the cells of the exported Excel file. For example, set background color. How to achieve this?
+
+How to customize the appearance of the cells in the exported Excel file?
+
+In Blazor WASM application customizing the exported file using RadSpreadProcessing is slow. How can I achieve a better performance?
+
+## Solution
+
+Handle the [OnAfterExport]({%slug grid-export-events%}#onafterexport) event of the Grid. It fires before the actual file is provided to the user. The `Stream` field of its event argument contains the output of the export as a `MemoryStream`.
+
+You can copy the bytes from this stream and import them in a [`RadSpreadStreamProcessing workbook`](https://docs.telerik.com/devtools/document-processing/libraries/radspreadstreamprocessing/model/workbook) to access and modify the cells.
+
+
+In WebAssembly applications customization of the exported Excel files is faster using the [RadSpreadStreamProcessing library](https://docs.telerik.com/devtools/document-processing/libraries/radspreadstreamprocessing/overview) compared to [using the RadSpreadProcessing one]({%slug grid-kb-custom-cell-formatting-with-radspreadprocessing%}).
+
+The example below targets Excel file export and customization. Same approach can be applied for CSV files.
+
+To customize the cell format of the exported file before it reaches the client, do the following:
+
+1. Handle the [OnAfterExport]({%slug grid-export-events%}#onafterexport) event of the Grid. The stream it provides is finalized, so that the resource does not leak. Its binary data, however, is available, so you can copy the stream bytes to a new `MemoryStream` instance.
```

Comment:
```suggestion
1. Handle the [`OnAfterExport`]({%slug grid-export-events%}#onafterexport) event of the Grid. The stream it provides is finalized so that the resource does not leak. Its binary data, however, is available so you can copy the stream bytes to a new `MemoryStream` instance.
```